### ISSUE TITLE
Assertion failures in debug mode

### ISSUE THREAD

**Issue by ghost** [ermia-db/ermia#6]
It appears that when building in debug mode, the assertion at [`base_txn_btree.cc:209`](https://github.com/ermia-db/ermia/blob/a3210d35c3c34f6663bddf6b8a87b4f1f4a4f4eb/base_txn_btree.cc#L209) always fails.

The problem doesn't exist when building in release mode (which I believe turn these assertions off). Is it still correct though?

**Comment by wangtzh** [comment]
I'm not seeing it in debug build... what's your config/parameters?

**Comment by ghost** [comment]
I configured my build with `cmake ../ -DCMAKE_BUILD_TYPE=Debug` (in a separate `build/` directory).

I then run it using

```bash
./ermia_SSI -verbose -benchmark tpcc -threads 12 -scale_factor 4 -seconds 5 \
            -null_log_device -log_data_dir=/dev/shm/yihehuang/ermia-log \
            -tmpfs_dir=/dev/shm -log_buffer_mb=128 -log_segment_mb=16384 \
            -parallel_loading -node_memory_gb=20
```

My system is Ubuntu 16.04 with the default gcc (5.4).

**Comment by wangtzh** [comment]
Thanks, I was able to reproduce it. Pull the latest to see if it works on your side.

It was the persistent back pointers on dummy versions (for deletion) causing the problem. These pointers (and thus the assertion) were added solely for log shipping to be able to dig up version in the log. So it won't affect standalone runs, which appears to be the case here. I'll take a deeper look later for both cases.
